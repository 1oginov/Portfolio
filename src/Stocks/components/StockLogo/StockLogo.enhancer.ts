import withStockLogoBySymbol from '../../enhancers/withStockLogoBySymbol';

export default withStockLogoBySymbol;
