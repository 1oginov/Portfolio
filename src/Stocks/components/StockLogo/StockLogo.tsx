import * as React from 'react';

import Progress from 'Shared/components/Progress';

// TODO: Tests.

export interface Props {
  className?: string;
  logo: string | null;
  logoProgress: boolean;
  size?: number;
}

const StockLogo: React.FunctionComponent<Props> = ({
  className, logo, logoProgress, size,
}: Props) => {
  if (logoProgress) {
    return <Progress className={className} />;
  }

  if (!logo) {
    return null;
  }

  return (
    <div className={className}>
      <img
        alt=""
        src={logo}
        style={{
          height: size,
          objectFit: 'contain',
          verticalAlign: 'top',
          width: size,
        }}
      />
    </div>
  );
};

StockLogo.defaultProps = {
  size: 40,
};

export default StockLogo;
