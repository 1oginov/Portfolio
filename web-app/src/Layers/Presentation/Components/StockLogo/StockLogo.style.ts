import { withStyles } from '@material-ui/core';

export const style = withStyles((theme) => ({
  root: {
    backgroundColor: theme.palette.background.paper,
    borderRadius: '10%',
    objectFit: 'contain',
  },
}));
