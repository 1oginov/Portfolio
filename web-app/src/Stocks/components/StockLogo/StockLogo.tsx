import classNames from 'classnames';
import * as React from 'react';

import { Progress } from 'Layers/Presentation/Progress';

export interface Props {
  classes: { [key: string]: string };
  className?: string;
  logo: string | null;
  logoProgress: boolean;
  size?: number;
}

const StockLogo: React.FunctionComponent<Props> = ({
  classes, className, logo, logoProgress, size,
}: Props) => {
  if (logoProgress) {
    return <Progress className={className} />;
  }

  if (!logo) {
    return null;
  }

  return (
    <img alt="" className={classNames(classes.root, className)} src={logo} style={{ height: size, width: size }} />
  );
};

StockLogo.defaultProps = {
  size: 40,
};

export default StockLogo;
