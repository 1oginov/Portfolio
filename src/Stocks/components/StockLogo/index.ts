import StockLogo from './StockLogo';
import enhance from './StockLogo.enhancer';
import style from './StockLogo.styles';

export default style(enhance(StockLogo));
