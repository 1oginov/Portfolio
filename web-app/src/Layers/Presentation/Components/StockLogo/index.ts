import {
  StockLogoEnhancer,
} from 'Layers/Behavior/Enhancers/StockLogoEnhancer/StockLogoEnhancer';

import { StockLogo as StockLogoComponent, StockLogoProps } from './StockLogo';
import { style } from './StockLogo.style';

export const StockLogo = style(StockLogoEnhancer<StockLogoProps>()(StockLogoComponent));
