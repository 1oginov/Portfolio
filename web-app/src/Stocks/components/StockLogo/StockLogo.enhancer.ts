import { compose } from 'recompose';

import withStockLogoBySymbol from '../../enhancers/withStockLogoBySymbol';

import { Props } from './StockLogo';

interface EnhancedProps {
  className?: string;
  size?: number;
  symbol: string;
}

export default compose<Props, EnhancedProps>(withStockLogoBySymbol);
