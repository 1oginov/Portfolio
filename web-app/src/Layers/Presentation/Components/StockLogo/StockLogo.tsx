import classNames from 'classnames';
import * as React from 'react';

import {
  StockLogoBySymbolEnhancerInputProps, StockLogoBySymbolEnhancerProps,
} from 'Layers/Behavior/Enhancers/StockLogoBySymbolEnhancer/StockLogoBySymbolEnhancer';
import { Progress } from 'Layers/Presentation/Components/Progress';

export interface StockLogoProps extends StockLogoBySymbolEnhancerInputProps {
  classes: { [key: string]: string };
  className?: string;
  size?: number;
}

type Props = StockLogoProps & StockLogoBySymbolEnhancerProps;

export const StockLogo: React.FunctionComponent<Props> = ({
  classes, className, logo, logoProgress, size,
}: Props) => {
  if (logoProgress) {
    return <Progress className={className} />;
  }

  if (!logo) {
    return null;
  }

  return (
    <img alt="" className={classNames(classes.root, className)} src={logo} style={{ height: size, width: size }} />
  );
};

StockLogo.defaultProps = {
  size: 40,
};
