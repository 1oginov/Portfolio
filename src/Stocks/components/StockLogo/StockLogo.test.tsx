import { shallow } from 'enzyme';
import classes from 'identity-obj-proxy';
import * as React from 'react';

import StockLogo from './StockLogo';

it('matches snapshot', () => {
  const wrapper = shallow(
    <StockLogo classes={classes} logo="https://example.com/stock/logo.png" logoProgress={false} />,
  );

  expect(wrapper).toMatchSnapshot();
});

it('matches snapshot when loading', () => {
  const wrapper = shallow(
    <StockLogo classes={classes} logo={null} logoProgress />,
  );

  expect(wrapper).toMatchSnapshot();
});

it('matches snapshot when logo is null', () => {
  const wrapper = shallow(
    <StockLogo classes={classes} logo={null} logoProgress={false} />,
  );

  expect(wrapper).toMatchSnapshot();
});
