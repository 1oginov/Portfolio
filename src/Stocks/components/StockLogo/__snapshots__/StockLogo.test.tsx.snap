// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<img
  alt=""
  className="root"
  src="https://example.com/stock/logo.png"
  style={
    Object {
      "height": 40,
      "width": 40,
    }
  }
/>
`;

exports[`matches snapshot when loading 1`] = `<Progress />`;

exports[`matches snapshot when logo is null 1`] = `""`;
